### PR TITLE
Add intuitive commands for local & global chat

### PR DESCRIPTION
**Pull request by walkersgaming**
Add intuitive commands for joining global & local chats. Easier to understand for newfriends that won't have an understanding of namelayer or groups yet.

/local
/localchat
/global
/globalchat

### CODE REVIEW

File: paper/src/main/java/vg/civcraft/mc/civchat2/commands/GlobalChat.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+	@Description("Enters global group chat or sends a message to global group chat")
+	@CommandCompletion("@CC_Groups @nothing")
+	public void execute(Player player, @Optional String chatMessage) {
+        String targetGroup = CivChat2.getInstance().getPluginConfig().getGlobalChatGroupName();
+		CivChat2Manager chatMan = CivChat2.getInstance().getCivChat2Manager();
+		GroupManager gm = NameAPI.getGroupManager();
+		boolean isGroupChatting = chatMan.getGroupChatting(player) != null;
+		Group group;
+		boolean defGroup = false;
+		if (targetGroup == null && chatMessage == null) {
```
Comment (awoo-civ) [inline paper/src/main/java/vg/civcraft/mc/civchat2/commands/GlobalChat.java]: These nested if statements are complex to the point where it's hard to keep track of the state. It should be possible to refactor these into a flatter, more understandable, structure.

Comment (grepsedawk) [reply paper/src/main/java/vg/civcraft/mc/civchat2/commands/GlobalChat.java]: Guard clauses and method extraction would be my first attempt to refactor this

File: paper/src/main/java/vg/civcraft/mc/civchat2/commands/GlobalChat.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+				}
+			} else {
+				player.sendMessage(String.format(ChatStrings.chatGroupNowChattingIn, group.getName()));
+				if (chatMan.getChannel(player) != null) {
+					chatMan.removeChannel(player);
+				}
+				chatMan.addGroupChat(player, group);
+			}
+		} else {
+			StringBuilder chatMsg = new StringBuilder();
```
Comment (Protonull) [inline paper/src/main/java/vg/civcraft/mc/civchat2/commands/GlobalChat.java]: Please don't :pray: copy over these StringBuilders :joy: See for yourself, they do nothing.

File: paper/src/main/java/vg/civcraft/mc/civchat2/commands/LocalChat.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+	@Description("Enter local chat")
+	public void execute(Player player, @Optional String message) {
+		CivChat2Manager chatMan = CivChat2.getInstance().getCivChat2Manager();
+		if (message == null) {
+			chatMan.removeChannel(player);
+			chatMan.removeGroupChat(player);
+			player.sendMessage(ChatStrings.chatMovedToGlobal);
+			return;
+		}
+		StringBuilder chatMsg = new StringBuilder();
```
Comment (Protonull) [inline paper/src/main/java/vg/civcraft/mc/civchat2/commands/LocalChat.java]: Likewise with this StringBuilder

**Comment by Diet-Cola** [comment]
Smart addition, I agree with the points above, something simple-r than citadels player states would be good here. If you want help with this in praxis, feel free to hit me up

File: paper/src/main/java/vg/civcraft/mc/civchat2/commands/GlobalChat.java [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+	@Syntax("[message]")
+	@Description("Enters global group chat or sends a message to global group chat")
+	@CommandCompletion("@CC_Groups @nothing")
+	public void execute(Player player, @Optional String chatMessage) {
+        String targetGroup = CivChat2.getInstance().getPluginConfig().getGlobalChatGroupName();
+		CivChat2Manager chatMan = CivChat2.getInstance().getCivChat2Manager();
+		GroupManager gm = NameAPI.getGroupManager();
+		boolean isGroupChatting = chatMan.getGroupChatting(player) != null;
+		Group group;
+		boolean defGroup = false;
```
Comment (awoo-civ) [inline paper/src/main/java/vg/civcraft/mc/civchat2/commands/GlobalChat.java]: This boolean is unused.

File: paper/src/main/java/vg/civcraft/mc/civchat2/commands/GlobalChat.java [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+		if (targetGroup == null && chatMessage == null) {
+			// Check if player is in groupchat and move them to normal chat
+			if (isGroupChatting) {
+				player.sendMessage(ChatStrings.chatMovedToGlobal);
+				chatMan.removeGroupChat(player);
+				return;
+			} else {
+				String defGroupName = gm.getDefaultGroup(player.getUniqueId());
+				if (defGroupName != null) {
+					group = GroupManager.getGroup(defGroupName);
```
Comment (awoo-civ) [inline paper/src/main/java/vg/civcraft/mc/civchat2/commands/GlobalChat.java]: This assignment is ineffectual - gets overridden by the one at line 50.

File: paper/src/main/java/vg/civcraft/mc/civchat2/commands/GlobalChat.java [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+import vg.civcraft.mc.namelayer.permission.PermissionType;
+
+public class GlobalChat extends BaseCommand {
+
+	@CommandAlias("global|globalchat")
+	@Syntax("[message]")
+	@Description("Enters global group chat or sends a message to global group chat")
+	@CommandCompletion("@CC_Groups @nothing")
+	public void execute(Player player, @Optional String chatMessage) {
+        String targetGroup = CivChat2.getInstance().getPluginConfig().getGlobalChatGroupName();
```
Comment (awoo-civ) [inline paper/src/main/java/vg/civcraft/mc/civchat2/commands/GlobalChat.java]: Style: This line is indented with spaces instead of tabs.

Comment (grepsedawk) [reply paper/src/main/java/vg/civcraft/mc/civchat2/commands/GlobalChat.java]: Spaces good

Comment (awoo-civ) [reply paper/src/main/java/vg/civcraft/mc/civchat2/commands/GlobalChat.java]: It's inconsistent ;)

Comment (grepsedawk) [reply paper/src/main/java/vg/civcraft/mc/civchat2/commands/GlobalChat.java]: I agree consistent good.

File: paper/src/main/java/vg/civcraft/mc/civchat2/commands/GlobalChat.java [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+import vg.civcraft.mc.namelayer.permission.PermissionType;
+
+public class GlobalChat extends BaseCommand {
+	@CommandAlias("global|globalchat")
+	@Syntax("[message]")
+	@Description("Enters global group chat or sends a message to global group chat")
+	@CommandCompletion("@CC_Groups @nothing")
+	
+	public void execute(Player player, @Optional String chatMessage) {
+		String globalGroupName = CivChat2.getInstance().getPluginConfig().getGlobalChatGroupName();
```
Comment (awoo-civ) [inline paper/src/main/java/vg/civcraft/mc/civchat2/commands/GlobalChat.java:25]: Variables should be defined as closely to the point of their first use as possible; the tighter the scope of a variable, the more understandable the code is.

**Comment by wingzero54** [comment]
Did all of @Protonull and @awoo-civ 's comments get addressed in the second commit @walkersgaming ?

If I'm understanding this properly, it also added handling so that running the command will also remove you from that chat when run? Ex: /local (add to local); run /local again (add to global)

**Comment by walkersgaming** [comment]
@wingzero54 I believe so yes.

No, if you run the command again it just says already in that group

**Pull request merged**